### ISSUE TITLE
S1854 reports false positive

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#4087]
### Description

S1854 reports false positive.

### Repro steps

```csharp
void Foo()
{
    int i = 42;

    try
    {
        i = int.Parse("i'll throw");
    }
    catch
    {
        Console.WriteLine(i);
    }
}
```

### Expected behavior

Diagnostic is not reported.

### Actual behavior

Diagnostic is reported (`remove the useless assignment...`). But if the assignment is removed, the code will no longer compile. 
### Related information

* SonarAnalyzer.CSharp 8.2.0-RC1

**Comment by hankovich** [comment]
Issue is probably related to the https://github.com/SonarSource/sonar-dotnet/issues/4085, but for `catch` blocks

**Comment by pavel-mikula-sonarsource** [comment]
Hi @hankovich,

Thank you for reporting this case, I can confirm it as False Positive. The root cause for this is already reported in #2600 that you can track so I'll close this one.
